schematic/store/synapse.py

```diff
@@ -538,6 +538,88 @@ def getDataTypeFromManifest(self, manifestId:str):
 
         return result_dict
 
+    def _get_files_under_datasets(self, datasetId: str) -> dict:
+        """retrieve file ids under a particular datasetId
+
+        Args:
+            datasetId (str): a dataset id 
+
+        Returns:
+            a dictionary that contains filename and entityid under a given datasetId
+        """
+        dataset_files = self.getFilesInStorageDataset(datasetId)
+        dataset_file_names_id_dict = {"Filename": [], "entityId": []}
+
+        for entity_id, file_name in dataset_files: 
+            dataset_file_names_id_dict["Filename"].append(file_name)
+            dataset_file_names_id_dict["entityId"].append(entity_id)
+
+        return dataset_file_names_id_dict
+
+    def add_entity_id_and_filename(self,  datasetId: str, manifest: pd.DataFrame) -> pd.DataFrame:
+        """add entityid and filename column to an existing manifest assuming entityId column is not already present
+
+        Args:
+            datasetId (str): dataset syn id
+            manifest (pd.DataFrame): existing manifest dataframe, assuming this dataframe does not have an entityId column and Filename column is present but completely empty 
+
+        Returns:
+            pd.DataFrame: returns a pandas dataframe 
+        """
+        # get file names and entity ids of a given dataset 
+        dataset_files_dict = self._get_files_under_datasets(datasetId)
+        new_files = pd.DataFrame(dataset_files_dict)
+        # concatenate the two dataframes together 
+        manifest = (
+                pd.concat([manifest, new_files], sort=False)
```

Comment:
Make sure that we already have `Filename` column we should account for it. The sorting should be done according to the sorting function in manifest generator.